### ISSUE TITLE
[🐞] `useStore` reactivity is broken if key isn't initialized in store object

### ISSUE THREAD

**Issue by jessezhang91** [BuilderIO/qwik#4174]
### Which component is affected?

Qwik Runtime

### Describe the bug

I am updating a store with an optional key. What I expect is downstream reactivity to occur based on the store's change. What actually is happening is no changes occur.

### Reproduction

https://stackblitz.com/edit/qwik-starter-dgqpxj?file=src%2Froutes%2Findex.tsx

### Steps to reproduce

Copied from reproduction link above:
```
  const signalWithInit = useSignal('init');
  const signalWithoutInit = useSignal<string | undefined>();

  const storeWithInit = useStore<{ value: string }>({ value: 'init' });
  const storeWithoutInit = useStore<{ value?: string }>({});

  useVisibleTask$(
    () => {
      signalWithInit.value = 'visible-task';
      signalWithoutInit.value = 'visible-task';

      storeWithInit.value = 'visible-task';
      storeWithoutInit.value = 'visible-task';
    },
    { strategy: 'document-ready' }
  );
  ```

`storeWithoutInit.value` will not trigger any downstream changes (e.g. rendering content or triggering `track`s).

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 931.25 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.13.0 - ~/.nvm/versions/node/v18.13.0/bin/node
    npm: 8.19.3 - ~/.nvm/versions/node/v18.13.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.92
    Firefox Developer Edition: 113.0
    Safari: 16.4
  npmPackages:
    @builder.io/qwik: ^1.1.1 => 1.1.1 
    @builder.io/qwik-city: ^1.1.1 => 1.1.1 
    undici: ^5.22.1 => 5.22.1 
    vite: 4.3.5 => 4.3.5
```

### Additional Information

Here's the same repro code as above but in v1.0.0:
https://stackblitz.com/edit/qwik-starter-odovoh?file=src%2Froutes%2Findex.tsx

It does render/function as expected there.

The workaround I'm going to attempt is to not have any optional keys in my store types and instead have it defaulted to `undefined` instead.

**Comment by cmbartschat** [comment]
This is a regression between 1.0.0 and 1.1.0 

[1.0.0 Playground](https://qwik.builder.io/playground/#version=1.0.0&buildMode=development&entryStrategy=hook&files=JTVCJTdCJTIycGF0aCUyMiUzQSUyMiUyRmFwcC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTVDbiUyMCUyMGNvbXBvbmVudCUyNCUyQyU1Q24lMjAlMjB1c2VTdG9yZSUyQyU1Q24lMjAlMjB1c2VWaXNpYmxlVGFzayUyNCUyQyU1Q24lN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJyUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGNvbXBvbmVudCUyNCgoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjBjb25zdCUyMHN0b3JlV2l0aG91dEluaXQlMjAlM0QlMjB1c2VTdG9yZSUzQyU3QiUyMHZhbHVlJTNGJTNBJTIwc3RyaW5nJTIwJTdEJTNFKCU3QiU3RCklM0IlNUNuJTVDbiUyMCUyMHVzZVZpc2libGVUYXNrJTI0KCU1Q24lMjAlMjAlMjAlMjAoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjBzdG9yZVdpdGhvdXRJbml0LnZhbHVlJTIwJTNEJTIwJ3Zpc2libGUtdGFzayclM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTJDJTVDbiUyMCUyMCUyMCUyMCU3QiUyMHN0cmF0ZWd5JTNBJTIwJ2RvY3VtZW50LXJlYWR5JyUyMCU3RCU1Q24lMjAlMjApJTNCJTVDbiU1Q24lMjAlMjByZXR1cm4lMjAoJTVDbiUyMCUyMCUyMCUyMCUzQyUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjBTdG9yZSUyMHclMkZvJTIwSW5pdCUzQSUyMCU3QnN0b3JlV2l0aG91dEluaXQudmFsdWUlN0QlNUNuJTIwJTIwJTIwJTIwJTNDJTJGJTNFJTVDbiUyMCUyMCklM0IlNUNuJTdEKSUzQiUyMiU3RCUyQyU3QiUyMnBhdGglMjIlM0ElMjIlMkZlbnRyeS5zZXJ2ZXIudHN4JTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHJlbmRlclRvU3RyaW5nJTJDJTIwdHlwZSUyMFJlbmRlck9wdGlvbnMlMjAlN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJTJGc2VydmVyJyUzQiU1Q25pbXBvcnQlMjAlN0IlMjBSb290JTIwJTdEJTIwZnJvbSUyMCcuJTJGcm9vdCclM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMChvcHRzJTNBJTIwUmVuZGVyT3B0aW9ucyklMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwcmVuZGVyVG9TdHJpbmcoJTNDUm9vdCUyMCUyRiUzRSUyQyUyMG9wdHMpJTNCJTVDbiU3RCU1Q24lMjIlN0QlMkMlN0IlMjJwYXRoJTIyJTNBJTIyJTJGcm9vdC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwQXBwJTIwZnJvbSUyMCcuJTJGYXBwJyUzQiU1Q24lNUNuZXhwb3J0JTIwY29uc3QlMjBSb290JTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlM0MlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDaGVhZCUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0N0aXRsZSUzRUhlbGxvJTIwUXdpayUzQyUyRnRpdGxlJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQyUyRmhlYWQlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDYm9keSUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0NBcHAlMjAlMkYlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDJTJGYm9keSUzRSU1Q24lMjAlMjAlMjAlMjAlM0MlMkYlM0UlNUNuJTIwJTIwKSUzQiU1Q24lN0QlM0IlNUNuJTIyJTdEJTVE)

[1.1.0 Playground](https://qwik.builder.io/playground/#version=1.1.0&buildMode=development&entryStrategy=hook&files=JTVCJTdCJTIycGF0aCUyMiUzQSUyMiUyRmFwcC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCJTVDbiUyMCUyMGNvbXBvbmVudCUyNCUyQyU1Q24lMjAlMjB1c2VTdG9yZSUyQyU1Q24lMjAlMjB1c2VWaXNpYmxlVGFzayUyNCUyQyU1Q24lN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJyUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGNvbXBvbmVudCUyNCgoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjBjb25zdCUyMHN0b3JlV2l0aG91dEluaXQlMjAlM0QlMjB1c2VTdG9yZSUzQyU3QiUyMHZhbHVlJTNGJTNBJTIwc3RyaW5nJTIwJTdEJTNFKCU3QiU3RCklM0IlNUNuJTVDbiUyMCUyMHVzZVZpc2libGVUYXNrJTI0KCU1Q24lMjAlMjAlMjAlMjAoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjBzdG9yZVdpdGhvdXRJbml0LnZhbHVlJTIwJTNEJTIwJ3Zpc2libGUtdGFzayclM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTJDJTVDbiUyMCUyMCUyMCUyMCU3QiUyMHN0cmF0ZWd5JTNBJTIwJ2RvY3VtZW50LXJlYWR5JyUyMCU3RCU1Q24lMjAlMjApJTNCJTVDbiU1Q24lMjAlMjByZXR1cm4lMjAoJTVDbiUyMCUyMCUyMCUyMCUzQyUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjBTdG9yZSUyMHclMkZvJTIwSW5pdCUzQSUyMCU3QnN0b3JlV2l0aG91dEluaXQudmFsdWUlN0QlNUNuJTIwJTIwJTIwJTIwJTNDJTJGJTNFJTVDbiUyMCUyMCklM0IlNUNuJTdEKSUzQiUyMiU3RCUyQyU3QiUyMnBhdGglMjIlM0ElMjIlMkZlbnRyeS5zZXJ2ZXIudHN4JTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHJlbmRlclRvU3RyaW5nJTJDJTIwdHlwZSUyMFJlbmRlck9wdGlvbnMlMjAlN0QlMjBmcm9tJTIwJyU0MGJ1aWxkZXIuaW8lMkZxd2lrJTJGc2VydmVyJyUzQiU1Q25pbXBvcnQlMjAlN0IlMjBSb290JTIwJTdEJTIwZnJvbSUyMCcuJTJGcm9vdCclM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMChvcHRzJTNBJTIwUmVuZGVyT3B0aW9ucyklMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwcmVuZGVyVG9TdHJpbmcoJTNDUm9vdCUyMCUyRiUzRSUyQyUyMG9wdHMpJTNCJTVDbiU3RCU1Q24lMjIlN0QlMkMlN0IlMjJwYXRoJTIyJTNBJTIyJTJGcm9vdC50c3glMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwQXBwJTIwZnJvbSUyMCcuJTJGYXBwJyUzQiU1Q24lNUNuZXhwb3J0JTIwY29uc3QlMjBSb290JTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlM0MlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDaGVhZCUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0N0aXRsZSUzRUhlbGxvJTIwUXdpayUzQyUyRnRpdGxlJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUzQyUyRmhlYWQlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDYm9keSUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0NBcHAlMjAlMkYlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTNDJTJGYm9keSUzRSU1Q24lMjAlMjAlMjAlMjAlM0MlMkYlM0UlNUNuJTIwJTIwKSUzQiU1Q24lN0QlM0IlNUNuJTIyJTdEJTVE)

**Comment by manucorporat** [comment]
We got a fix!

**Comment by Zeiver** [comment]
Short question,
since it's about broken reactivity and it was introduced with v1.1.0,
could it fix the context reactivity problem too? https://discord.com/channels/842438759945601056/1106601727458824302/1106601727458824302

**Comment by manucorporat** [comment]
@Zeiver the discord thread does not have much source code i can use to reproduce

**Comment by Zeiver** [comment]
https://stackblitz.com/edit/node-xenwez?file=src/components/starter/example/example.tsx
So I only created a 1.1.0 version, but it's the same problem.
If I give the store an initial value even if it's undefined for { text?: string } then it works.
@manucorporat 

And I gave up on stackblitz, tried to create anything in that and nothing is working.
Create a new repo, pushed my stuff there and loaded it with stackblitz.
